### PR TITLE
[PM-2516][PM-8298] Add password protected export on browser/Export managed collection

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The introduction of a password-protected export was a great feature addition, but unfortunately until now was only available on web and the CLI. 

This work adds the following features:
- Password protected exports are now available on browser
- Using the same visual component as the web vault for a more modern look (Follow-up PR to further modernize the UI with the Extension refresh following soon)
- With the recent introduction of flexible collections, user can now manage their assigned collection. As already possible on the web-vault members of an organization can now also export their managed collection from the browser extension

## Code changes

- **Create export-browser component using shared recipe -** https://github.com/bitwarden/clients/commit/d9b6dfe5026f1694334898eb172d2f1bd09fef3c
  - Create new export component that uses the shared export.component from @bitwarden/vault-export-ui
- **Update imports within AppModule -** https://github.com/bitwarden/clients/commit/f73b53aee0d50cd95edc5673242de912b6029da3
- **Switch to route to the new component -** https://github.com/bitwarden/clients/commit/f740590815c3f8fe3bd9b6eb15663f6dd25547e7
- **Add missing entries to messages.json -** https://github.com/bitwarden/clients/commit/aa10835fd71469a6103ff717c001e331cc0d4963
- **Delete old export.component -** https://github.com/bitwarden/clients/commit/ce5a4af46a0129445b64cc29f3bf9688eab192b1

### Before
before_export

### After
after_export1
after_export2
after_org-export

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/browser/src/_locales/en/messages.json [hunk 1]
```diff
@@ -751,12 +751,39 @@
...
+  "filePassword": {
+    "message": "File password"
+  },
+  "exportPasswordDescription": {
+    "message": "This password will be used to export and import this file"
+  },  
+  "accountRestrictedOptionDescription": {
+    "message": "Use your account encryption key, derived from your account's username and Master Password, to encrypt the export and restrict import to only the current Bitwarden account."
+  },
+  "passwordProtectedOptionDescription": {
```
Comment (audrey-jensen) [inline apps/browser/src/_locales/en/messages.json:775]: 🤔 Comparing with #9286, should this also have a "passwordProtected" message?

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -2134,6 +2161,18 @@
...
   },
+  "exportingOrganizationVaultTitle": {
+    "message": "Exporting organization vault"
+  },
+  "exportingOrganizationVaultDesc": {
+    "message": "Only the organization vault associated with $ORGANIZATION$ will be exported. Items in individual vaults or other organizations will not be included.",
+    "placeholders": {
+      "organization": {
+        "content": "$1",
+        "example": "My Org Name"
```
Comment (audrey-jensen) [inline apps/browser/src/_locales/en/messages.json:2172]: ⛏️ I might provide something that sounds like an org name for the example.

"ACME Moving Co."

File: apps/browser/src/tools/popup/settings/export/export-browser.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    RouterLink,
+    JslibModule,
+    DialogModule,
+    AsyncActionsModule,
+    ButtonModule,
+    ExportComponent,
+  ],
+})
+export class ExportBrowserComponent {
+  protected disabled = false;
```
Comment (audrey-jensen) [inline apps/browser/src/tools/popup/settings/export/export-browser.component.ts:23]: ⛏️ I'd document the protected members of this method.

**Review by audrey-jensen — COMMENTED** [review]
🤔 non-blocking concern
⛏️ nitpick

**Review by audrey-jensen — DISMISSED** [review]

File: apps/browser/src/_locales/en/messages.json (hunk 1 above)
Comment (djsmith85) [reply apps/browser/src/_locales/en/messages.json:817]: The key was already present in this file: https://github.com/bitwarden/clients/blob/5dd722414376034f3a84d2ebe587c77ba6b74754/apps/browser/src/_locales/en/messages.json#L1839

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (djsmith85) [reply apps/browser/src/_locales/en/messages.json]: Addressed with bee06acf742c68421389acd010b9d8d8d237ecd0

File: apps/browser/src/tools/popup/settings/export/export-browser.component.ts (hunk 3 above)
Comment (djsmith85) [reply apps/browser/src/tools/popup/settings/export/export-browser.component.ts:27]: Addressed with 0dd1e0e5896756c9aca04a16601a62987fa6d887

**Review by audrey-jensen — APPROVED** [review]

**Pull request merged**